test/e2e/es/mutation_test.go

```diff
@@ -160,6 +160,25 @@ func TestMutationSecondMasterSetDown(t *testing.T) {
 	RunESMutation(t, b, mutated)
 }
 
+// TestMutationRollingDownscaleCombination combines a rolling update with scale down operation.
+func TestMutationRollingDownscaleCombination(t *testing.T) {
+	b := elasticsearch.NewBuilder("test-combined-upgrade-downscale").
+		WithESMasterNodes(3, elasticsearch.DefaultResources).
+		WithNamedESDataNodes(2, "data-1", elasticsearch.DefaultResources).
+		WithNamedESDataNodes(3, "data-2", elasticsearch.DefaultResources)
```

Comment:
should we make it a bit less resource intensive so we can run it on smaller k8s clusters? 1 dedicated master, 1 data node in the first set, 2 in the second one?